File: tks/RepositoriesAdapters/src/main/kotlin/mappers/AccessionNumberMapper.java
package mappers;

import exceptions.TypeValidationFailedException;
import model.values.AccessionNumber;
import org.mapstruct.Mapper;

@Mapper
public interface AccessionNumberMapper {

    default String toValue(AccessionNumber accessionNumber){
        return accessionNumber.getValue();
    }

    default AccessionNumber toAccessionNumber(String value) throws TypeValidationFailedException {
        return new AccessionNumber(value);
    }
}

File: tks/RepositoriesAdapters/src/main/kotlin/mappers/AccessionNumberMapper.kt
package mappers

import model.values.AccessionNumber
import kotlin.Throws
import exceptions.TypeValidationFailedException
import org.mapstruct.Mapper

@Mapper
interface AccessionNumberMapper {
    @JvmDefault
    fun toValue(accessionNumber: AccessionNumber): String? {
        return accessionNumber.value
    }

    @JvmDefault
    @Throws(TypeValidationFailedException::class)
    fun toAccessionNumber(value: String?): AccessionNumber? {
        return AccessionNumber(value)
    }
}